lib/modules/dosomething/dosomething_global/dosomething_global.module

```diff
@@ -16,6 +16,35 @@ define('DOSOMETHING_GLOBAL_DEFAULT_LANG_CODE', 'en-global');
 function dosomething_global_init() {
   global $user;
 
+  // Redirect people hitting the homepage to there local version
+  if (drupal_is_front_page() && is_numeric(arg(1))) {
+
+    // Cannot use arg() as it doesn't return path prefixes
+    $path = request_path();
+    $args = explode('/', $path);
+
+    // Only apply logic to Global homepage.
+    if (!empty($path) && $args[0] != 'node') {
+      return;
+    }
```

Comment:
unfortunately it had to be wild...

all of these are supporting edge cases / weird things I found. (eg: even if the URL is just / the arg 0 and 1 params still return the homepage node). The homepage is a weird thing, the comments point out which part handles what.